src/index.md

```diff
@@ -14,15 +14,40 @@ in the officially maintained [Rust Book](https://doc.rust-lang.org/book/).
 To read the book about gdnative (Godot 3 binding), follow [this link](../gdnative-book).
 
 
+## The basics
+
+A **library** is in its most basic form is a **collection of functions**. Godot has several of its own built-in libraries, but like most other programs made as a creation tool, it also has the capacity to access collections of libraries made outside of Godot. Those are called **extensions**, and like most other software, they usually come in separate files -- typically in the form of file extensions such as `.so` if you're on Linux, `.dll` if you're on Windows, or `.dylib` on MacOS.
+
+The way that GDExtension work is that you basically expose these files to your Godot project by using a `.gdextension` file pointing to the thing. Such file will have all the information of where that file is and which version of Godot it should be compatible with.
```

Comment:
> The way that GDExtension work is that you basically expose [...] pointing to the thing.

This sounds quite sloppy; I would avoid unnecessarily vague terms like "thing" (or maybe even "basically") and directly describe what happens, in a concise way:

> You can expose your extension to Godot using a `.gdextension` file, which contains information about the dynamic library's location and the minimum required Godot version.